cdap-docs/build.sh

```diff
@@ -20,283 +20,210 @@
-  run_command javadocs
-  run_command docs-cli
-  run_command docs-github-part ${ARG_2} 
-  stash_github_zip 
-  run_command docs-web-part ${ARG_2} 
-  restore_github_zip
-  set_display_version
-  display_messages_file
-  warnings=$?
-  cleanup_messages_file
+}
+  
+function display_end_title() {
   echo "--------------------------------------------------------"
-  ring_bell "Completed \"build all\""
+  echo "Completed \"$1\""
   echo "========================================================"
   echo "========================================================"
   echo
-  exit ${warnings}
 }
 
-function build_docs() {
-  local warnings
-  local doc_type=${1}
-  local source_path=${ARG_2}
-  local javadocs="${WITHOUT}"
-  local cli_docs="${WITHOUT}"
-  if [ "${doc_type}" == "${GITHUB}" -o "${doc_type}" == "${WEB}" ]; then
-    javadocs="${WITH}"
-    cli_docs="${WITH}"
-  fi
-  if [ "${doc_type}" == "${WEB_ONLY}" -o "${doc_type}" == "${GITHUB_ONLY}" ]; then
-    cli_docs="${WITH}"
-  fi
-  if [ "${doc_type}" == "${DOCS_LOCAL}" ]; then
-    DOCS_LOCAL="true"
-    export DOCS_LOCAL
-  fi
+function display_end_title_bell() {
+  echo "--------------------------------------------------------"
+  ring_bell "Completed \"$1\""
   echo "========================================================"
   echo "========================================================"
-  echo "Building \"${doc_type}\""
-  echo "(${cli_docs} CLI docs, ${javadocs} Javadocs)"
-  echo "--------------------------------------------------------"
   echo
-  if [[ "${doc_type}" != "${DOCS}" && "${doc_type}" != "${DOCS_OUTER}" ]]; then
-    clean_targets
-  fi
-  clear_messages_set_messages_file
-  if [ "${doc_type}" != "${DOCS_OUTER}" ]; then
-    run_command docs-first-pass ${source_path}
-  fi
-  if [ "${doc_type}" == "${DOCS}" -o "${doc_type}" == "${DOCS_OUTER}" ]; then
-    if [ "${javadocs}" != "${WITH}" -o "${cli_docs}" != "${WITH}" ]; then
-      check_build_rst
-    fi
-    build_docs_outer_level ${source_path}
-    copy_docs_inner_level
-  else
-    clear_messages_set_messages_file
-  fi
-  if [ "${javadocs}" == "${WITH}" -o "${cli_docs}" == "${WITH}" ]; then
-    if [ "${javadocs}" == "${WITH}" ]; then
-      run_command javadocs
-    fi
-    if [ "${cli_docs}" == "${WITH}" ]; then
-      run_command docs-cli
-    fi
-  else
-    check_build_rst
-  fi
-  if [ "${doc_type}" == "${GITHUB}" -o "${doc_type}" == "${GITHUB_ONLY}" ]; then
-    run_command docs-github-part ${source_path}
-  elif [ "${doc_type}" == "${WEB}" -o "${doc_type}" == "${WEB_ONLY}" ]; then
-    run_command docs-web-part ${source_path}
-  fi
-  set_display_version
-  display_messages_file
+}
+
+function build_docs_set() {
+  local warnings
+  local title="Building Docs Set"
+  display_start_title ${title}
+  
+  clean_targets
+  clear_messages_file
+  build_docs_first_pass
+  clear_messages_file
+  build_javadocs docs
+  build_docs_cli
+  build_docs_second_pass
+  build_docs_package
```

Comment:
OK, but this code isn't written in a way for that to make sense. A failure in one of these functions will cause cascading failures/problems. They're not written as fully independent units of work. Therefore, the script should fail fast, rather than continue generating false-positive errors (along with actual errors, yes) and using up build resources.